public/app/features/alerting/unified/components/rule-editor/notificaton-preview/NotificationPreview.test.tsx

```diff
@@ -210,7 +210,7 @@ describe('NotificationPreview', () => {
     await waitFor(() => {
       expect(ui.loadingIndicator.query()).not.toBeInTheDocument();
     });
-    await userEvent.click(ui.seeDetails.get());
+    await userEvent.click(await ui.seeDetails.find());
```

Comment:
Did you work out _why_ these needed to change to finding them async? What was it that changed with Redux that made it behave slightly differently? 🤔

Reply:
yes and no. i don't know _exactly_, but the timing of when things render and run in the test has changed slightly.

this checks is now running too quickly:
```
    await waitFor(() => {
      expect(ui.loadingIndicator.query()).not.toBeInTheDocument();
    });
```
so doesn't actually do anything. it runs before the spinner even appears. you can test this by changing it to
```
    await waitFor(() => {
      expect(ui.loadingIndicator.query()).toBeInTheDocument();
    });
    await waitFor(() => {
      expect(ui.loadingIndicator.query()).not.toBeInTheDocument();
    });
    await userEvent.click(ui.seeDetails.get());
```
then the `get` works correctly. instead, i've just removed the waiting for the loading spinner to disappear as it's kind of an antipattern to me anyway. better to wait for something to appear than wait for something to disappear.

Reply:
Yeah agreed, this pattern is safer/better anyway :+1: Thanks! I imagine its something subtle in the upgrade, but this way of asserting still makes sense for me